### PR TITLE
Replace deprecated 'az configure' with 'az config' and update syntax

### PR DESCRIPTION
**Pull request by dksimpson**
This work is for story [1844036](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1844036). 

Replace instances of the deprecated 'az configure' with 'az config' and update syntax in examples.

### CODE REVIEW

**Comment by dksimpson** [comment]
Hi @dbradish-microsoft, this PR is ready for review.

**Comment by dbradish-microsoft** [comment]
@jiasli , This is for Azure DevOps work item [Deprecated az configure](https://dev.azure.com/mseng/TechnicalContent/_workitems/edit/1844036/).  Can you please give these changes a quick review for any additional information you feel is needed for `az config`?  Thank you.

File: docs-ref-conceptual/release-notes-azure-cli.md [hunk 1]
```diff
@@ -5459,7 +5459,7 @@ Version 2.0.42
 ### SQL
 
 * Fixed 'The provided resource group name ... did not match the name in the Url' error when specifying elastic pool name for `sql db copy` and `sql db replica create` commands
-* Allow configuring default sql server by executing `az configure --defaults sql-server=<name>`
```
Comment (dbradish-microsoft) [inline docs-ref-conceptual/release-notes-azure-cli.md:5462]: @dksimpson , please take release-notes-azure-cli.md out of this PR.  This file should only be modified by the engineering team.  It is a reflection of a "point in time" i.e. history.

**Review by dbradish-microsoft — CHANGES_REQUESTED** [review]

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 1 above)
Comment (dksimpson) [reply docs-ref-conceptual/release-notes-azure-cli.md]: Done

File: docs-ref-conceptual/format-output-azure-cli.md [hunk 2]
```diff
@@ -189,10 +189,10 @@ KBDemo020
 
 ## Set the default output format
 
-Use the interactive `az configure` command to set up your environment and establish default settings for output formats. The default output format is `json`.
+Use the interactive `az config` command to set up your environment and establish default settings for output formats. The default output format is `json`.
```
Comment (jiasli) [inline docs-ref-conceptual/format-output-azure-cli.md:192]: `az config` is not interactive.

```suggestion
Use the `az config set` command to set up your environment and establish default settings for output formats. The default output format is `json`.
```

File: docs-ref-conceptual/format-output-azure-cli.md [hunk 3]
```diff
@@ -189,10 +189,10 @@ KBDemo020
...
 
 ```azurecli-interactive
-az configure
+az config get
```
Comment (jiasli) [inline docs-ref-conceptual/format-output-azure-cli.md:195]:
```suggestion
azure config set core.output=<format>
```

File: docs-ref-conceptual/param-persist-howto.md [hunk 4]
```diff
@@ -72,15 +72,15 @@ az config param-persist show
 
 ## Persisted parameter and global variable comparison
 
-There are two Azure CLI commands that can be used to default parameter values: `az configure` and `az config param-persist`.  Use the `az configure` command to specify _global variables_ such as group, location, or web.  Use `az param-persist` to specify _local default values_ unique to your workload.  Stored values are used by the CLI in place of required arguments.
+There are two Azure CLI commands that can be used to default parameter values: `az config set defaults` and `az config param-persist`.  Use the `az config set defaults` command to specify _global variables_ such as group, location, or web.  Use `az param-persist` to specify _local default values_ unique to your workload.  Stored values are used by the CLI in place of required arguments.
```
Comment (jiasli) [inline docs-ref-conceptual/param-persist-howto.md:75]: Better to be more explicit:

```suggestion
There are two Azure CLI commands that can be used to default parameter values: `az config set defaults` and `az config param-persist`.  Use the `az config set defaults.<option>=<value>` command to specify _global variables_ such as group, location, or web.  Use `az param-persist` to specify _local default values_ unique to your workload.  Stored values are used by the CLI in place of required arguments.
```

File: docs-ref-conceptual/param-persist-howto.md [hunk 5]
```diff
@@ -72,15 +72,15 @@ az config param-persist show
...
 
 > [!Important]
 > Persisted parameters override global context values.
 >
 
 | Reference | Scope | Set | Use
 |-|-|-|-|
-[az configure](/cli/azure/reference-index#az_configure) | Scoped globally across the CLI | Set explicitly using `az configure --defaults` | Use for settings such as logging, data collection, and default argument values
+[az config set defaults](/cli/azure/config) | Scoped globally across the CLI | Set explicitly using `az config set defaults` | Use for settings such as logging, data collection, and default argument values
```
Comment (jiasli) [inline docs-ref-conceptual/param-persist-howto.md:83]: `default` is a positional argument, not part of the command name.

```suggestion
[az config set defaults.<option>=<value>](/cli/azure/config) | Scoped globally across the CLI | Set explicitly using `az config set defaults.<option>=<value>` | Use for settings such as logging, data collection, and default argument values
```

Actually the functionality of `az config` is a super set of `az config`. The comparison between `az config` and `az config param-persist` is becomes meaningless.

File: docs-ref-conceptual/param-persist-howto.md [hunk 6]
```diff
@@ -157,4 +157,4 @@ Even with a global variable set for resource group with a value of `myGlobalVari
 ## See also
 
 * [Tutorial: Use persisted parameter with sequential Azure CLI commands](param-persist-tutorial.md)
-* [Azure CLI Configuration using az configure](azure-cli-configuration.md)
+* [Azure CLI Configuration using az config](azure-cli-configuration.md)
```
Comment (jiasli) [inline docs-ref-conceptual/param-persist-howto.md:160]:
```suggestion
* [Azure CLI Configuration using `az config`](azure-cli-configuration.md)
```

File: docs-ref-conceptual/reference-types-and-status.md [hunk 7]
```diff
@@ -21,7 +21,7 @@ The Azure CLI syntax is a combination of references, commands, and parameters.
 
 | Azure Service | Reference | Reference subservice | Command | Full reference command | Parameter Examples
 |-|-|-|-|-|-|
-| Azure CLI | [az configure](/cli/azure/reference-index#az_configure) | | | az configure | --defaults, --list-default, --scope
+| Azure CLI | [az config](/cli/azure/config) | | | az config | get, param-persist, set, unset
```
Comment (jiasli) [inline docs-ref-conceptual/reference-types-and-status.md:24]: `get, param-persist, set, unset` are not parameters, but subcommands.

**Comment by dksimpson** [comment]
Hi @jiasli, I've updated the articles and incorporated your review comments. Ready for final review.
FYI @dbradish-microsoft

**Comment by dbradish-microsoft** [comment]
@dksimpson , please fix the new merge conflict, then we'll merge.

**Comment by dksimpson** [comment]
Hi @dbradish-microsoft, I just resolved the merge conflict.

**Pull request merged**